### ISSUE TITLE
Setuptools does not use version from versioneer

### ISSUE THREAD

**Issue by torfsen** [python-versioneer/python-versioneer#102]
When I run `setup.py sdist`, versioneer does set the correct version, but somehow it is not picked up by setuptools. In particular, setuptools uses a version string that is not compliant with PEP440. Take a look at the last 4 lines:

```
$ ./setup.py sdist
running sdist
running egg_info
writing dependency_links to src/clusterpolate.egg-info/dependency_links.txt
writing requirements to src/clusterpolate.egg-info/requires.txt
writing src/clusterpolate.egg-info/PKG-INFO
writing top-level names to src/clusterpolate.egg-info/top_level.txt
reading manifest file 'src/clusterpolate.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'src/clusterpolate.egg-info/SOURCES.txt'
warning: cmd_sdist: standard file not found: should have one of README, README.rst, README.txt

running check
creating clusterpolate-0.1.0-0.ge7cb0e8.dirty
creating clusterpolate-0.1.0-0.ge7cb0e8.dirty/src
creating clusterpolate-0.1.0-0.ge7cb0e8.dirty/src/clusterpolate
creating clusterpolate-0.1.0-0.ge7cb0e8.dirty/src/clusterpolate.egg-info
making hard links in clusterpolate-0.1.0-0.ge7cb0e8.dirty...
hard linking LICENSE -> clusterpolate-0.1.0-0.ge7cb0e8.dirty
hard linking MANIFEST.in -> clusterpolate-0.1.0-0.ge7cb0e8.dirty
hard linking README.md -> clusterpolate-0.1.0-0.ge7cb0e8.dirty
hard linking requirements.txt -> clusterpolate-0.1.0-0.ge7cb0e8.dirty
hard linking setup.cfg -> clusterpolate-0.1.0-0.ge7cb0e8.dirty
hard linking setup.py -> clusterpolate-0.1.0-0.ge7cb0e8.dirty
hard linking versioneer.py -> clusterpolate-0.1.0-0.ge7cb0e8.dirty
hard linking src/clusterpolate/__init__.py -> clusterpolate-0.1.0-0.ge7cb0e8.dirty/src/clusterpolate
hard linking src/clusterpolate/_version.py -> clusterpolate-0.1.0-0.ge7cb0e8.dirty/src/clusterpolate
hard linking src/clusterpolate.egg-info/PKG-INFO -> clusterpolate-0.1.0-0.ge7cb0e8.dirty/src/clusterpolate.egg-info
hard linking src/clusterpolate.egg-info/SOURCES.txt -> clusterpolate-0.1.0-0.ge7cb0e8.dirty/src/clusterpolate.egg-info
hard linking src/clusterpolate.egg-info/dependency_links.txt -> clusterpolate-0.1.0-0.ge7cb0e8.dirty/src/clusterpolate.egg-info
hard linking src/clusterpolate.egg-info/requires.txt -> clusterpolate-0.1.0-0.ge7cb0e8.dirty/src/clusterpolate.egg-info
hard linking src/clusterpolate.egg-info/top_level.txt -> clusterpolate-0.1.0-0.ge7cb0e8.dirty/src/clusterpolate.egg-info
copying setup.cfg -> clusterpolate-0.1.0-0.ge7cb0e8.dirty
Writing clusterpolate-0.1.0-0.ge7cb0e8.dirty/setup.cfg
UPDATING clusterpolate-0.1.0-0.ge7cb0e8.dirty/src/clusterpolate/_version.py
set clusterpolate-0.1.0-0.ge7cb0e8.dirty/src/clusterpolate/_version.py to '0.1.0+0.ge7cb0e8.dirty'
Creating tar archive
removing 'clusterpolate-0.1.0-0.ge7cb0e8.dirty' (and everything under it)
```

As you can see, versioneer reports the version as `0.1.0+0.ge7cb0e8.dirty` (which should be PEP440 compliant), but setuptools uses `clusterpolate-0.1.0-0.ge7cb0e8.dirty` instead (which is not). Note that running `setup.py sdist` twice does not change anything. Running `setup.py bdist` has the same problem:

```
$ ./setup.py bdist
running bdist
running bdist_dumb
running build
running build_py
UPDATING build/lib.linux-x86_64-2.7/clusterpolate/_version.py
set build/lib.linux-x86_64-2.7/clusterpolate/_version.py to '0.1.0+0.ge7cb0e8.dirty'
installing to build/bdist.linux-x86_64/dumb
running install
running install_lib
creating build/bdist.linux-x86_64/dumb
creating build/bdist.linux-x86_64/dumb/home
creating build/bdist.linux-x86_64/dumb/home/torf
creating build/bdist.linux-x86_64/dumb/home/torf/projects
creating build/bdist.linux-x86_64/dumb/home/torf/projects/clusterpolate
creating build/bdist.linux-x86_64/dumb/home/torf/projects/clusterpolate/venv
creating build/bdist.linux-x86_64/dumb/home/torf/projects/clusterpolate/venv/lib
creating build/bdist.linux-x86_64/dumb/home/torf/projects/clusterpolate/venv/lib/python2.7
creating build/bdist.linux-x86_64/dumb/home/torf/projects/clusterpolate/venv/lib/python2.7/site-packages
creating build/bdist.linux-x86_64/dumb/home/torf/projects/clusterpolate/venv/lib/python2.7/site-packages/clusterpolate
copying build/lib.linux-x86_64-2.7/clusterpolate/__init__.py -> build/bdist.linux-x86_64/dumb/home/torf/projects/clusterpolate/venv/lib/python2.7/site-packages/clusterpolate
copying build/lib.linux-x86_64-2.7/clusterpolate/_version.py -> build/bdist.linux-x86_64/dumb/home/torf/projects/clusterpolate/venv/lib/python2.7/site-packages/clusterpolate
byte-compiling build/bdist.linux-x86_64/dumb/home/torf/projects/clusterpolate/venv/lib/python2.7/site-packages/clusterpolate/__init__.py to __init__.pyc
byte-compiling build/bdist.linux-x86_64/dumb/home/torf/projects/clusterpolate/venv/lib/python2.7/site-packages/clusterpolate/_version.py to _version.pyc
running install_egg_info
running egg_info
writing dependency_links to src/clusterpolate.egg-info/dependency_links.txt
writing requirements to src/clusterpolate.egg-info/requires.txt
writing src/clusterpolate.egg-info/PKG-INFO
writing top-level names to src/clusterpolate.egg-info/top_level.txt
reading manifest file 'src/clusterpolate.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'src/clusterpolate.egg-info/SOURCES.txt'
Copying src/clusterpolate.egg-info to build/bdist.linux-x86_64/dumb/home/torf/projects/clusterpolate/venv/lib/python2.7/site-packages/clusterpolate-0.1.0_0.ge7cb0e8.dirty-py2.7.egg-info
running install_scripts
creating /home/torf/projects/clusterpolate/dist
Creating tar archive
removing 'build/bdist.linux-x86_64/dumb' (and everything under it)
```

This is on a fresh installation of versioneer 0.15. My `setup.py` contains

```
...
import versioneer
...
setup(
    ...
    version=versioneer.get_version(),
    cmdclass=versioneer.get_cmdclass(),
    ...
)
```

Here is my `setup.cfg`:

```
[metadata]
description-file = README.md

[versioneer]
VCS = git
style = pep440
versionfile_source = src/clusterpolate/_version.py
versionfile_build = clusterpolate/_version.py
tag_prefix =

[coverage:run]
omit = src/clusterpolate/_version.py
```

And my `MANIFEST.in`:

```
include requirements.txt
include README.md
include LICENSE

include versioneer.py
include src/clusterpolate/_version.py
```

Finally, my package's `__init__.py` contains

```
from __future__ import (absolute_import, division, print_function,
                        unicode_literals)

...

# Versioneer
from ._version import get_versions
__version__ = get_versions()['version']
del get_versions
```

I'm using Python 2.7.6 inside a virtualenv.

The project [is on GitHub](https://github.com/torfsen/clusterpolate) if you need to take a closer look.

**Comment by effigies** [comment]
@torfsen I recognize this is very old, but can you confirm whether this is still an issue?

**Comment by torfsen** [comment]
@effigies Hah, this issue is ancient, I'd totally forgotten about it :laughing: 

Since I haven't been working with versioneer or the affected project in quite some time I only did a quick check, but it seems that the issue is fixed in `0.18`. So I will go ahead and close the ticket.

Thanks for reaching out!
